ext/stackprof/stackprof.c

```diff
@@ -113,9 +129,14 @@ stackprof_start(int argc, VALUE *argv, VALUE self)
 	sigaction(mode == sym_wall ? SIGALRM : SIGPROF, &sa, NULL);
 
 	timer.it_interval.tv_sec = 0;
-	timer.it_interval.tv_usec = NUM2LONG(interval);
+	timer.it_interval.tv_usec = NUM2INT(interval);
 	timer.it_value = timer.it_interval;
 	setitimer(mode == sym_wall ? ITIMER_REAL : ITIMER_PROF, &timer, 0);
+
+        if (_stackprof.debug) {
+            printf("started with interval %d (%ld sec %d usec)\n",
+                NUM2INT(interval), timer.it_interval.tv_sec, timer.it_interval.tv_usec);
```

Comment:
(This highlights the separate issue of https://github.com/tmm1/stackprof/issues/125)